bimmer_connected/vehicle/remote_services.py

```diff
@@ -73,91 +103,56 @@ def __init__(self, vehicle: "MyBMWVehicle"):
-        """Trigger the vehicle to flash its headlights.
-
-        A state update is NOT triggered after this, as the vehicle state is unchanged.
-        """
-        _LOGGER.debug("Triggering remote light flash")
-        return await self.trigger_remote_service(Services.LIGHT_FLASH)
-
-    async def trigger_remote_door_lock(self) -> RemoteServiceStatus:
-        """Trigger the vehicle to lock its doors.
-
-        A state update is triggered after this, as the lock state of the vehicle changes.
-        """
-        _LOGGER.debug("Triggering remote door lock")
-        result = await self.trigger_remote_service(Services.DOOR_LOCK)
-        await self._trigger_state_update()
-        return result
-
-    async def trigger_remote_door_unlock(self) -> RemoteServiceStatus:
-        """Trigger the vehicle to unlock its doors.
-
-        A state update is triggered after this, as the lock state of the vehicle changes.
-        """
-        _LOGGER.debug("Triggering remote door unlock")
-        result = await self.trigger_remote_service(Services.DOOR_UNLOCK)
-        await self._trigger_state_update()
-        return result
-
-    async def trigger_remote_horn(self) -> RemoteServiceStatus:
-        """Trigger the vehicle to sound its horn.
+    async def trigger_remote_service(
+        self, service_id: Services, params: Optional[Dict] = None, data: Any = None, refresh: bool = False
+    ) -> RemoteServiceStatus:
+        """Trigger a remote service and wait for the result."""
 
-        A state update is NOT triggered after this, as the vehicle state is unchanged.
-        """
-        _LOGGER.debug("Triggering remote horn sound")
-        return await self.trigger_remote_service(Services.HORN)
-
-    async def trigger_charge_now(self) -> RemoteServiceStatus:
-        """Trigger the vehicle to start charging.
+        _LOGGER.debug(
+            "Triggering remote service %s with params %s and data %s",
+            repr(service_id),
+            params,
+            json.dumps(data, cls=MyBMWJSONEncoder) if data else None,
+        )
 
-        A state update is NOT triggered after this, as the vehicle state is unchanged.
-        """
-        _LOGGER.debug("Triggering charge now")
-        result = await self.trigger_remote_service(Services.CHARGE_NOW)
-        await self._trigger_state_update()
-        return result
+        # Check if service requires a specific url and add all required parameters
+        url = SERVICE_URLS.get(service_id, REMOTE_SERVICE_URL)
+        url = url.format(vin=self._vehicle.vin, service_type=service_id.value)
 
-    async def trigger_remote_air_conditioning(self) -> RemoteServiceStatus:
-        """Trigger the air conditioning to start.
+        # Trigger service and get event id
+        async with MyBMWClient(self._account.config, brand=self._vehicle.brand) as client:
+            response = await client.post(
+                url, params=params, json=json.dumps(data, cls=MyBMWJSONEncoder) if data else None
```

Comment:
Check whether data is already a valid json object, otherwise serialize it:

`url, params=params, json=data if is_json(json.dumps(data)) else json.dumps(data, cls=MyBMWJSONEncoder) if data else None`